docs/docs/guides/creating-your-own-interpretation.md

```diff
@@ -61,13 +61,53 @@ class MemoizeNegativeProperty(Interpretation, alias="memoize_negative"):
 Again, it will be valuable to read the API details on nodestream's model.
 The above code leverages the aforementioned `InterpretationContext` as well as `DesiredIngest` and `PropertySet`.
 
-## Make sure your module is imported
+## Register Your Interpretation
 
-Wherever you have your class defined, nodestream needs to know that its something that should be imported. To do
-so, add your module to the imports section of your `nodestream.yaml` file. For example:
+
+Interpretations are registered via the [entry_points](https://setuptools.pypa.io/en/latest/userguide/entry_point.html#entry-points-for-plugins) API of a Python Package. Specifically, the `entry_point` named `interpretations` inside of the `nodestream.plugins` group is loaded. Every Value Provider is expected to be a subclass of `nodestream.interpreting:Interpretation` as directed above. 
+
+The `entry_point` should be a module that contains at least one Value Provider class. At runtime, the module will be loaded and all classes that inherit from `nodestream.interpreting:Interpretation` will be registered. The `alias` attribute of the class will be used as as the name of the tag used in the yaml pipeline.
+
+Depending on how you are building your package, you can register your Value Provider plugin in one of the following ways:
+
+=== "pyproject.toml"
```

Comment:
These example sections are redundant, should it be abstracted out and referred to in one place?  

Maybe have a "See examples" file and link to it?

Reply:
Yeah I agree. I will see if there is a good way to de-duplicate this that works nicely in the docs.

Reply:
I tried the suggestion out, but it was a little cumbersome to follow a step-by-step guide over multiple pages. Do you have a strong preference to eliminate the redundancy? I found it "familiar" that each guide follows a similar structure and gets you from start to finish.